### PR TITLE
Fix NULL set of fkeys in QgsFeatureListModel

### PR DESCRIPTION
**Pull request by signedav**
## Description
The function `setExtraIdentifierValuesToNull` did not set the indexes. Leaded to not updating widgets like RelationReference:
bug_relation_reference

Now `setExtraIdentifierValuesToNull` calls `setExtraIdentifierValues( QVariantList() )` what makes it obsolete. The relation reference widgets calls `setExtraIdentifierValues( QVariantList() )` directly.

The tests are improved for this case.

### CODE REVIEW

**Comment by m-kuhn** [comment]
@troopa81 you should be fairly familiar with this code IIRC, can I ask you a review?

**Comment by nyalldawson** [comment]
@signedav 

> "using of setIdentifierValues( QVariantList() ) when setting to null to set the index to null index"

Can I suggest that this commit message could be made more useful by adding a description of the bug its fixing? Right now the only way to determine what this has fixed is by the screencast in the PR message, but that's lost as soon as the PR is merged and the change just becomes a single line entry in the git log...

File: src/core/qgsfeaturefiltermodel.h [hunk 1]
```diff
@@ -196,8 +196,9 @@ class CORE_EXPORT QgsFeatureFilterModel : public QAbstractItemModel
      * Allows specifying one value that does not need to match the filter criteria but will
      * still be available in the model as NULL value(s).
      * \since QGIS 3.10
+     * \deprecated since QGIS 3.12 use setExtraIdentifierValues( QVariantList() ) instead
      */
-    void setExtraIdentifierValuesToNull();
+    Q_DECL_DEPRECATED void setExtraIdentifierValuesToNull() SIP_DEPRECATED;
```
Comment (troopa81) [inline src/core/qgsfeaturefiltermodel.h]: Is it really necessary to deprecate this method? I mean, the way you fix it should work now and it doesn't harm to keep it.

Same comment worth for setIdentifierValuesToNull

File: src/gui/editorwidgets/qgsrelationreferencewidget.cpp [hunk 2]
```diff
@@ -391,12 +391,12 @@ void QgsRelationReferenceWidget::deleteForeignKeys()
   }
   else
   {
-    mComboBox->setIdentifierValuesToNull();
+    mComboBox->setIdentifierValues( QVariantList() );
```
Comment (troopa81) [inline src/gui/editorwidgets/qgsrelationreferencewidget.cpp]: We could keep calling the setIdentifierValuesToNull since you fixed it no?

**Comment by troopa81** [comment]
> @troopa81 you should be fairly familiar with this code IIRC, can I ask you a review?

Sure. Not completly aware of this part of the code, but I take a look

**Comment by troopa81** [comment]
@signedav I made some minor comments but it seems OK to me

File: src/core/qgsfeaturefiltermodel.h (hunk 1 above)
Comment (signedav) [reply src/core/qgsfeaturefiltermodel.h]: Okay, I removed it again. And used always this function on setting to null.

**Review by troopa81 — APPROVED** [review]
Failed test is unrelated to this PR (pointlocatortest, I'll take a look). It looks fine to me

**Comment by signedav** [comment]
Thanks @troopa81

**Pull request merged**